qvisor/src/runc/runtime/vm_type.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021 Quark Container Authors / 2018 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,x
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+pub mod noncc;
+pub mod resources;
+
+use std::sync::Arc;
+use kvm_ioctls::{Kvm, VmFd};
+use crate::{arch::vm::vcpu::ArchVirtCpu, elf_loader::KernelELF, qlib::common::Error, runc::runtime};
+use runtime::{vm::VirtualMachine, loader::Args};
+
+pub trait VmType: std::fmt::Debug {
```

Comment:
I don't see why a supertrait is needed here, this trait doesn't provide any default impl that uses the Debug trait.